plugins/amqp_witness_plugin/amqp_witness_plugin.cpp

```diff
@@ -46,14 +46,14 @@ void amqp_witness_plugin::plugin_startup() {
 
    my->rqueue = std::make_unique<reliable_amqp_publisher>(my->amqp_server, my->exchange, my->routing_key, witness_data_file_path, "eosio.node.witness_v0");
 
-   app().get_plugin<witness_plugin>().add_on_witness_sig([my=my](const chain::block_state_ptr& bsp, const chain::signature_type& sig) {
+   app().get_plugin<witness_plugin>().add_on_witness_sig([my=my.get()](const chain::block_state_ptr& bsp, const chain::signature_type& sig) {
       my->rqueue->post_on_io_context([my, bsp, sig]() {
          my->rqueue->publish_message(std::make_pair(bsp->header.action_mroot, sig));
```

Comment:
This can be executed after the destructor of rqueue starts.  "If an object of a standard library type is accessed, and ... the access does not happen before the end of the object’s lifetime, the behavior is undefined unless otherwise specified." [res.on.objects]